### ISSUE TITLE
SFT loss

### ISSUE THREAD

**Issue by lujiarui-iie** [dmis-lab/RetPO#2]
Hello, this is an excellent piece of work. I am using four 24GB 4090 GPUs to perform SFT on qwen1.5-0.5B, and I noticed that the loss remains at 0.000000. Is this reasonable?
```
{'loss': '0.000000', 'learning_rate': 1.9998178635346865e-05, 'epoch': 0.01}                                                                                                       
{'loss': '0.000000', 'learning_rate': 1.9992715204861295e-05, 'epoch': 0.01}                                                                                                       
{'loss': '0.000000', 'learning_rate': 1.9983611698723126e-05, 'epoch': 0.02}                                                                                                       
{'loss': '0.000000', 'learning_rate': 1.9970871433093214e-05, 'epoch': 0.02}                                                                                                       
{'loss': '0.000000', 'learning_rate': 1.9954499048905464e-05, 'epoch': 0.03}                                                                                                       
{'loss': '0.000000', 'learning_rate': 1.9934500510176242e-05, 'epoch': 0.04}
```

Additionally, could you help clarify whether it is reasonable that 4x24GB of GPU memory is not enough to fine-tune a 3B model with LoRA? (even batchsize=1)

Looking forward to your reply, and thank you in advance!

**Comment by gankim** [comment]
Hi, thanks for your interest in our work—and apologies for the delayed response.

We haven’t encountered a zero loss issue like the one you described, so this case is indeed unusual. One likely cause could be related to the tokenizer. Since we customized parts of the preprocessing pipeline (see: [preprocess.py](https://github.com/dmis-lab/RetPO/blob/main/retpo_qr/cqr/prep/preprocess.py)), you may need to adjust the preprocessing logic to support models other than LLaMA, which we originally used.

It would also be helpful to double-check that your dataset is correctly formatted before feeding it into the training process. Verifying this step often helps catch tokenization or input-label alignment issues early on.

Thanks again!
